src/coordinates.rs

```diff
@@ -0,0 +1,206 @@
+//! Extension traits and structs that provide generic access to the coordinates of
+//! elliptic curve points.
+//!
+//! Coordinates are meaningless without the context of the curve equation that constrains
+//! them. To safely expose them in a generic context, we use extension traits to restrict
+//! the scope of the generic curve parameter; this ensures that the code can only be used
+//! with curve implementations that explicitly expose their use of a specific curve model.
+
+use subtle::{Choice, ConditionallySelectable, CtOption};
+
+use crate::CurveAffine;
+
+//
+// Twisted Edwards curve
+//
+
+/// An affine elliptic curve point on a twisted Edwards curve
+/// $a \cdot x^2 + y^2 = 1 + d \cdot x^2 \cdot y^2$.
+pub trait TwistedEdwardsPoint: CurveAffine + Default + ConditionallySelectable {
```

Comment:
Hmm, following the existing `group` crate conventions (given that we use the one trait to represent both the curve and points on the curve), we should probably name this:
```suggestion
pub trait TwistedEdwardsCurve: CurveAffine + Default + ConditionallySelectable {
```